### ISSUE TITLE
Github: Logout fails, left in bugged state

### ISSUE THREAD

**Issue by DJRHails** [atom/github#2469]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: <https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode>
    * Followed all applicable steps in the debugging guide: <https://flight-manual.atom.io/hacking-atom/sections/debugging/>
    * Checked the FAQs on the message board for common solutions: <https://discuss.atom.io/c/faq>
    * Checked that your issue isn't already filed: <https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom>
    * Checked that there is not already an Atom package that provides the described functionality: <https://atom.io/packages>

### Description
Git tab works absolutely fine, and loads in the correct user profile based on ssh keys installed.
![Screenshot from 2020-05-27 14-09-32](https://user-images.githubusercontent.com/10828202/83023222-880c5d80-a024-11ea-882e-dae6e4b0a06d.png)
Github tab has unknown person in user profile
![Screenshot from 2020-05-27 14-09-52](https://user-images.githubusercontent.com/10828202/83023233-8b074e00-a024-11ea-9021-dcd436df5ced.png)
If I try and publish a new repo, I get a consistent loading... dropdown. I can see that the component is getting re-rendered in the inspector but it never gets past this step. 
![Screenshot from 2020-05-27 14-10-07](https://user-images.githubusercontent.com/10828202/83023248-8e023e80-a024-11ea-821b-a2d4bdc192a8.png)
Trying to logout fails.
![Peek 2020-05-27 14-12](https://user-images.githubusercontent.com/10828202/83023278-965a7980-a024-11ea-8d88-02727aac178e.gif)

I've also tried disabling the token, and credential.helper but to no success.

### Steps to Reproduce
1. Create a new folder, open atom with `atom .`
2. Open git tab / github tab.
3. Attempt to `Github: Logout`.

**Expected behavior:**
I should be able to logout and log back in with original credentials. Or at worse have some way to hard purge the credential system.

**Actual behavior:**
Cannot logout

**Reproduces how often:**
Once in this state it always occurs, can still manage just using git, but would like to utilise the github tab. I suspect it has something to do with the git credential cache management but I don't understand how the normal git tab would then work.

Have since reinstalled from scratch (purging .atom did not help) which allowed me to re-authenticate with a token.

### Versions
```bash
Δ atom --version             
aAtom    : 1.47.0
Electron: 5.0.13
Chrome  : 73.0.3683.121
Node    : 12.0.0

Δ apm --version             
apm  2.4.5
npm  6.2.0
node 10.2.1 x64
atom 1.47.0
python 2.7.18rc1
git 2.25.1
```

### Additional Information

```
Host github.com
  IdentityFile /home/<DJRHails>/.ssh/github
  LogLevel ERROR
```

**Comment by TehWan** [comment]
I'm experiencing the same issue on Ubuntu 20.04.1. I've revoked the permissions on GitHub and tried countless times to logout from within Atom, but there's no change. I still get the "loading..." username. I've tried searching for the token in the configuration files, but searching for "github" yields far too many results.

```
$ apm --version
apm  2.5.0
npm  6.14.5
node 10.20.1 x64
atom 1.49.0
python 3.8.2
git 2.25.1
```
Package github is at version 0.34.2. Disabling the package, restarting atom, any combination of that, nothing works. The logout command does nothing.

**Comment by TehWan** [comment]
In that state, clicking on the "Change repository with the dropdown" button causes the package to crash. My github package is now completely unusable.

```
TypeError [ERR_INVALID_ARG_TYPE] [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type object
    at validateString (internal/validators.js:105)
    at Object.normalize (path.js:1003)
    at GithubTabHeaderView.renderWorkDirs (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at GithubTabHeaderView.render (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at vi (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at ui (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at ps (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at fk (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at Zj (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at Lj (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at /snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14
    at e.unstable_runWithPriority (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at fg (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at ig (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at jg (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at Rg (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at Object.enqueueSetState (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at GithubTabHeaderController.E.setState (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at /snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14
    at Function.simpleDispatch (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:11)
    at Emitter.emit (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:11)
    at WorkdirContextPool.set (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at GithubPackage.getNextContext (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:11)
    at GithubPackage.updateActiveContext (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:11)
    at Task.execute (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
    at AsyncQueue.processTask (/snap/atom/258/usr/share/atom/resources/app/static/<embedded>:14)
```

**Comment by TehWan** [comment]
I somehow fixed it on my computer. I tried deleting the `~/.atom` folder and reinstalling atom, to no avail. I reinstated my `~/.atom` folder.
Using the command line, I created a git repository and pushed it to github, then loaded Atom on that folder, and now it's asking me to login. The issue seems to be when a git repository is initialized but does not have a remote.

**Comment by TehWan** [comment]
I can confirm the same issue happens on my laptop with a brand new project, and it happens only when a git repository does not have a remote that is on GitHub, or even when there is no git repository. Adding a remote that is on GitHub, I then get the login screen.

**Comment by FourInchKnife** [comment]
i have this same problem. i deleted my old github account and made a new one and even that didnt fix it.

this issue makes atom completely unusable. im currently using windows

**Comment by PaperFlu** [comment]
This situation is extremely similar to mine #2497 . I agree with what @TehWan said, this issue seems to be when a repository does not have a remote. The information I could add is in the issue I mentioned.

So, it is probably not that you can't log out, the real situation may be like that you have already logged out, but in a repository which doesn't have a remote.

I didn't mention it in the issue, but somehow I found that the account in the Git tab and the account in the GitHub tab are actually separated. That's why I think you have already logged out, in the GitHub tab. See my issue if you are looking for more.

To fix that, you can open an empty atom window, clone a remote repository, and the login button should appear. Login and back to your working project, everything should work like charm.

**Comment by DJRHails** [comment]
@PaperFlu Suspect these are related too.
However, while the repository I demoed with doesn't have a remote, I first discovered the issue on a private github repo that did have a remote (although it was pulled down via command line) which is worth keeping in mind.

**Comment by FourInchKnife** [comment]
If you want to fix this just `git commit` and `git push` any changes you’ve made then delete the clone.

After that, `git clone` your repo and then open atom there with `atom .`

**Comment by smashwilson** [comment]
This looks like the issue I fixed in #2572, where we weren't always showing the GitHub login prompt appropriately. In the meantime, as a workaround, opening a repository that always has a remote should allow you to login correctly.

Sorry for the inconvenience, and thank you for the report 🙇🏻
